### PR TITLE
Publish package

### LINKED ISSUE #266: Push first version to PyPi
For that, create a github workflow that does it when tagged, and depends on all other workflows passing.

### PR DESCRIPTION
**Pull request by zain-sohail**
Now we have the capability of prereleasing to pypi whenever we push to main. And releasing when we push a tag. 
The release version can be arbitrarily set by the person who releases (by defining the tag). 
The package is distributed as sed-processor and can be found on https://pypi.org/project/sed-processor/

There are four steps to what the workflow does:
- Workflow runs prerelease job on every push to main branch and a release job on every tag push.
  - prerelease example run: https://github.com/OpenCOMPES/sed/actions/runs/6897868624
  - release example run: https://github.com/OpenCOMPES/sed/actions/runs/6898553103
- A publish to pypi job is executed on every successful prerelease or release job
- And if publish is successful, the version is also updated in the pyproject.toml file and pushed to main branch (so we are always upto date)
  - By default, a workflow run that is triggered by a push event will not trigger another push event when it pushes changes to the repository so we don't have worries for recursive runs.

Necessary changes to be made before merging:

- [x] change workflow to make this run on main
- [x] let other workflows run before merging
- [x] fix environment to allow main

closes #266 

Changes made:
- docs dependencies now have a seperate section in toml file. This way they are not packaged for users (only developers need to generate docs), just like dev-dependencies aren't packaged
- updated license for current year
- fixed shields
- updated name of documentation workflow so we can get a nice shield
- updated readme
- **changed documentation structure** to allow readme and tutorials first

Improvements possible:
- simplify workflow with the sed-processor paths (but not sure how possible that is)
- further enhance the maintain file
- add the hextof tutorial in documentation

### CODE REVIEW

**Comment by steinnymir** [comment]
This looks great! 
Do I understand correctly, that a patch vesrsion is automatically updated when we merge a PR to main?
For example, now we are at `0.1.4`. Will the next merge bump us to `0.1.5`?

And for releases, we should instead change the minor or even major, i.e. `0.1.4` -> `0.2.0`?

**Comment by zain-sohail** [comment]
> This looks great! Do I understand correctly, that a patch vesrsion is automatically updated when we merge a PR to main?
> 
> For example, now we are at `0.1.4`. Will the next merge bump us to `0.1.5`? or how do pre-release versions look like?

prerelease are great because users don't get to install them unless they explicitly say. They look like this:
https://pypi.org/project/sed-processor/#history
so `0.1.4` -> `0.1.5a0` -> `0.1.5a1` 
and then when we feel ready to make a minor or patch release, we can do so manually. 
by saying
`git tag v0.1.5`
`git push origin v0.1.5`

which would make it:  `0.1.5a1` ->  `0.1.5` 

From semantic versioning
```
MAJOR version when you make incompatible API changes
MINOR version when you add functionality in a backward compatible manner
PATCH version when you make backward compatible bug fixes
```
Instead of prereleasing, we can also patch everytime, but I think it's better to do it manually just because if things get broken in the automated pipeline, at least we don't have a release.


> And for releases, we should instead change the minor or even major, i.e. `0.1.4` -> `0.2.0`?
yess. we probably want to do 1.0.0 when all the important PRs are merged.

File: pyproject.toml
Comment (steinnymir) [inline pyproject.toml:8]: is it possible to add the main authors explicitly, @zain-sohail @rettigl and @steinnymir ? It would be nice if we all appear in the pypi page. I'm not sure if this is happening here though.

File: README.md
Comment (steinnymir) [inline README.md:43]: I would add a "bleeding edge" installation, where we propose to run `run pip install -e .` in the main repo folder to install it interactively. This would be the preferred way when working in beamtimes with developers, as certainly a new branch with quick implementation of feature requests will spawn

**Review by steinnymir — COMMENTED** [review]
This looks good! I'll test installing it from pypi now, and approve if this works!

**Comment by steinnymir** [comment]
> so `0.1.4` -> `0.1.5a0` -> `0.1.5a1`

perfect! this I love this!

And yes, I think once most of the current open PRs are merged we can bump to `1.0.0`!

**Comment by rettigl** [comment]
This looks great, yes!
There is now the email address sed-processor@mpes.science, which for the moment links to my email, but the idea is to get an email list for this. So I suggest to use if for publishing as disussed in #183.

**Comment by rettigl** [comment]
We can make the pipy badge link to the pipy page.

File: pyproject.toml
Comment (zain-sohail) [reply pyproject.toml:8]: Of course it is. But I think we decided to use this at some point. Can also just put the names
I can easily add everyone to the maintainers in PyPi. I just need your PyPI account email addresses.

File: README.md
Comment (zain-sohail) [reply README.md:43]: I think then we can come up with three categories of people using sed:

- Users
- Contributors
- Maintainers (This doesn't have to be in README but only in documentation)

And contributors can have the bleeding edge installation. However, if they want to create docs or run tests locally, they can't without using poetry (but that's not an issue I guess due to all the github checks we have)

I am not sure how contributors work so could you write a small section and I can just push that?

**Review by steinnymir — APPROVED** [review]
It worked, I got it installed on maxwell with python 3.10 by simply creating a conda environment, and running pip install sed-processor.

The only problem is we have really a LOT of dependencies! 

LGTM!

**Comment by zain-sohail** [comment]
> It worked, I got it installed on maxwell with python 3.10 by simply creating a conda environment, and running pip install sed-processor.
> 
> The only problem is we have really a LOT of dependencies!

Yes, I also noticed it and we really should see if we can remove some.

File: README.md
Comment (zain-sohail) [reply README.md:43]: I have pushed some changes to README. You can edit if you wish

File: pyproject.toml
Comment (zain-sohail) [reply pyproject.toml]: invited

File: README.md
Comment (steinnymir) [reply README.md:58]: This is good for now. I will see what the feedback is next week and modify accordingly.

File: pyproject.toml
Comment (steinnymir) [reply pyproject.toml]: accepted. thanks!

**Comment by rettigl** [comment]
> Yes, I also noticed it and we really should see if we can remove some.

A lot of them come from pynxtools...

**Pull request merged**
